packages/visx-xychart/src/hooks/usePointerEventHandlers.ts

```diff
@@ -0,0 +1,117 @@
+  const { width, height, horizontal, dataRegistry, xScale, yScale } = useContext(DataContext);
+
+  const handlePointerMoveOrUp = useCallback(
+    (params?: HandlerParams) => {
+      const { svgPoint, event } = params || {};
+
+      const pointerParamsByKey: { [dataKey: string]: PointerEventParams<Datum> } = {};
+
+      // nearest Datum across all dataKeys, if relevant
+      let nearestDatumPointerParams: PointerEventParams<Datum> | null = null;
+      let nearestDatumDistance = Infinity;
+
+      if (params && event && svgPoint && width && height && xScale && yScale) {
+        const considerAllKeys =
+          dataKey === POINTER_EVENTS_NEAREST || dataKey === POINTER_EVENTS_ALL;
+
+        // find nearestDatum for relevant dataKey(s)
+        (considerAllKeys
+          ? dataRegistry?.keys()
+          : Array.isArray(dataKey)
+          ? dataKey
+          : [dataKey]
+        )?.forEach(key => {
+          const entry = dataRegistry?.get(key);
+          if (entry) {
+            const nearestDatum = (horizontal ? findNearestDatumY : findNearestDatumX)({
+              data: entry.data,
+              height,
+              point: svgPoint,
+              width,
+              xAccessor: entry.xAccessor,
+              xScale,
+              yAccessor: entry.yAccessor,
+              yScale,
+            });
+            if (nearestDatum) {
+              pointerParamsByKey[key] = { key, svgPoint, event, ...nearestDatum };
+
+              // compute nearest Datum if not emitting events for all keys\
+              if (dataKey === POINTER_EVENTS_NEAREST) {
+                const distance = Math.sqrt(
+                  (nearestDatum.distanceX ?? Infinity ** 2) +
+                    (nearestDatum.distanceY ?? Infinity ** 2),
+                );
+                nearestDatumPointerParams =
+                  distance < nearestDatumDistance
+                    ? pointerParamsByKey[key]
+                    : nearestDatumPointerParams;
+                nearestDatumDistance = Math.min(nearestDatumDistance, distance);
+              }
+            }
+          }
+        });
+
+        const pointerParams: (null | PointerEventParams<Datum>)[] =
+          dataKey === POINTER_EVENTS_NEAREST
+            ? [nearestDatumPointerParams]
+            : dataKey === POINTER_EVENTS_ALL || Array.isArray(dataKey)
+            ? Object.values(pointerParamsByKey)
+            : [pointerParamsByKey[dataKey]];
+
+        pointerParams.forEach(p => {
+          if (p?.event.type === 'pointerup' && onPointerUp) {
+            onPointerUp(p);
+          } else if (p?.event.type === 'pointermove') {
+            if (onPointerMove) onPointerMove(p);
+          }
+        });
+      }
+    },
+    [dataKey, xScale, yScale, width, height, horizontal, onPointerMove, onPointerOut, onPointerUp],
+  );
+
+  const handlePointerOut = useCallback(
+    (params?: HandlerParams) => {
+      if (params && onPointerOut) onPointerOut(params.event);
```

Comment:
because we have to always use the same # of hooks, we can't conditionally create this and have to do the check in the handler. we conditionally return the handler below, tho.